File: Algorithms/134. Gas Station/swift/solution.swift
class Solution {
    func canCompleteCircuit(_ gas: [Int], _ cost: [Int]) -> Int {
        var startStation = 0
        var totalTank = 0
        var currTank = 0
        for i in 0..<gas.count {
            totalTank += gas[i] - cost[i]
            currTank += gas[i] - cost[i]
            if currTank < 0 {
                startStation = i + 1
                currTank = 0
            }
        }
        return totalTank >= 0 ? startStation : -1
    }
}

File: Algorithms/134. Gas Station/golang/solution.go
func canCompleteCircuit(gas []int, cost []int) int {
	totalTank := 0
	currTank := 0
	startStation := 0

	for i := 0; i < len(gas); i++ {
		totalTank += gas[i] - cost[i]
		currTank += gas[i] - cost[i]

		if currTank < 0 {
			startStation = i + 1
			currTank = 0
		}
	}

	if totalTank >= 0 {
		return startStation
	}
	return -1
}
